lib/commands/context.js

```diff
@@ -0,0 +1,461 @@
+  let el = await this.findElement('name', 'URL');
+  this.implicitWaitMs = oldImpWait;
+  await this.nativeTap(el.ELEMENT);
+  let els = await this.findElements('name', 'Address');
+  let addressEl = els[els.length - 1].ELEMENT;
+  await this.setValueImmediate(addressEl, initialUrl);
+  el = await this.findElement('name', 'go');
+  await this.nativeTap(res.ELEMENT);
+  await this.navToViewWithTitle(/.*/i);
+  // Waits for page to finish loading.
+  await this.remote.pageUnload();
+};
+
+extensions.navToViewThroughFavorites = async function () {
+  logger.debug('We are on iOS7+ simulator: clicking apple button to get into a webview');
+  let oldImpWait = this.implicitWaitMs;
+  this.implicitWaitMs = 7000; // wait 7s for apple button to exist
+  try {
+    let el = await this.findElement('xpath', '//UIAScrollView[1]/UIAButton[1]');
+  } catch (err) {
+    let msg = 'Could not find button to click to get into webview. ' +
+              'Proceeding on the assumption we have a working one.';
+    logger.error(msg);
+    this.implicitWaitMs = oldImpWait;
+    return await this.navToViewWithTitle(/.*/i);
+  }
+  this.implicitWaitMs = oldImpWait;
+  try {
+    let res = await this.nativeTap(el.ELEMENT);
+    if (res.status !== 0) {
+      throw new Error('Failed!');
+    }
+  } catch (err) {
+    let msg = 'Could not click button to get into webview. ' +
+              'Proceeding on the assumption we have a working one.';
+    logger.error(msg);
+  }
+  await this.navToViewWithTitle(/apple/i);
+};
+
+extensions.navToViewWithTitle = async function (titleRegex) {
+  logger.debug('Navigating to most recently opened webview');
+  let start = Date.now();
+  let spinTime = 500;
+  let spinHandles = async () => {
+    let res;
+    try {
+      res = await this.getLatestWebviewContextForTitle(titleRegex);
+    } catch (err) {
+      throw new Error(`Could not navigate to webview! Err: ${err.message}`);
+    }
+    if (res) {
+      let latestWindow = res;
+      logger.debug(`Picking webview '${latestWindow}'`);
+      await this.setContext(latestWindow);
+      await this.remote.cancelPageLoad();
+      return;
+    } else {
+      if ((Date.now() - start) < 90000) {
+        logger.warn("Could not find any webviews yet, refreshing/retrying");
+        if (this.isRealDevice() || !this.caps.safari) {
+          await B.delay(spinTime);
+          return await spinHandles();
+        }
+        let element;
+        try {
+          element = await this.findUIElementOrElements('accessibility id', 'ReloadButton', '', false);
+        } catch (err) {
+          logger.warn('Could not find reload button, continuing');
+          await B.delay(spinTime);
+          return await spinHandles();
+        }
+        try {
+          await this.nativeTap(element.ELEMENT);
+        } catch (err) {
+          logger.warn('Could not click reload button, continuing');
+          await B.delay(spinTime);
+          return await spinHandles();
+        }
+      } else {
```

Comment:
like a previous comment i made, since this is a tiny little block that ends control of this function, why don't we reverse the boolean and put this up top? then we need fewer indented brace blocks :-)